From d85709dd0387b0228b20f911cd0436c4f9f84a4f Mon Sep 17 00:00:00 2001
From: Bin Han <bin.b.han@intel.com>
Date: Thu, 26 Nov 2015 11:09:08 +0800
Subject: [PATCH 22/28] ov8858: check vcm_driver->init is valid

Check vcm_driver->init is valid before calling it, because
on BTNS we do not really have a VCM therefore no init function
for VCM.

Change-Id: I176b4613d8c53eba1864ca91d2909584b0b333fe
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-9279
Original-Tracked-On: #H1504101742
Signed-off-by: Bin Han <bin.b.han@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/86714
Approver: Tian Shu Qiu <tian.shu.qiu@intel.com>
Validation-Android: Wei Tao <wei.tao@intel.com>
Build: Dessert M COHO (SC)
Build: Dessert M BXT RVP (SC)
Integrator: WeiX Meng <weix.meng@intel.com>
Reviewed-by: WeiX Meng <weix.meng@intel.com>
Signed-off-by: Yang Fan F <fan.f.yang@intel.com>
Reviewed-on: https://android.intel.com:443/442545
---
 drivers/external_drivers/camera/drivers/media/i2c/ov8858.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/external_drivers/camera/drivers/media/i2c/ov8858.c b/drivers/external_drivers/camera/drivers/media/i2c/ov8858.c
index 06f8400..f9748d4 100644
--- a/drivers/external_drivers/camera/drivers/media/i2c/ov8858.c
+++ b/drivers/external_drivers/camera/drivers/media/i2c/ov8858.c
@@ -1683,7 +1683,10 @@ static int __update_ov8858_device_settings(struct ov8858_device *dev,
 	else
 		return -ENODEV;
 
-	return dev->vcm_driver->init(&dev->sd);
+	if (dev->vcm_driver && dev->vcm_driver->init)
+		return dev->vcm_driver->init(&dev->sd);
+
+	return 0;
 }
 
 static int ov8858_s_config(struct v4l2_subdev *sd,
-- 
1.9.1

